File: 2022/Day 01/1.hs
import Data.List
import Data.Function
import Text.Printf

main = sortBy (flip compare).map (\xs -> sum [read x |x <- xs, not $ null x])
    .groupBy (on (==) null).lines <$> readFile "input.txt" >>= \(elves :: [Int]) ->
    let solve = sum.(`take` elves) in on(printf "Silver: %d\nGold: %d\n") solve 1 3

File: 2022/Day 01/1.py
Totals = [0] 

for ln in open('input.txt'):
    if (Str := ln.strip()):
        Totals[-1] += int(Str)
    else:
        Totals.append(0)

Totals = sorted(Totals)
print('silver:', Totals[-1])
print('gold:', sum(Totals[-3:]))
